assets/js/blocks/cart-checkout-shared/view-switcher/utils.ts

```diff
@@ -0,0 +1,59 @@
+/**
+ * External dependencies
+ */
+import { select } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import type { View } from './types';
+
+export const getView = (
+	viewName: string,
+	views: View[]
+): View | undefined => {
+	return views.find( ( view ) => view.view === viewName );
+};
+
+const defaultView = {
+	views: [],
+	currentView: '',
+	viewClientId: '',
+};
+
+export const findParentBlockEditorViews = (
+	clintId: string,
+	maxDepth = 10,
+	currentDepth = 0
+): {
+	views: View[];
+	currentView: string;
+	viewClientId: string;
+} => {
+	const depth = currentDepth + 1;
+
+	if ( depth > maxDepth ) {
+		return defaultView;
+	}
+
+	const { getBlockAttributes, getBlockRootClientId } =
+		select( 'core/block-editor' );
+	const rootId = getBlockRootClientId( clintId ) as string;
```

Comment:
```suggestion
	const rootId = getBlockRootClientId( clientId ) as string;
```

Reply:
To avoid `as string` here, should we return if it's not one? I feel like `as string` is a bit of a backdoor in TS and should be avoided if possible.

That said, if we _know_ for sure it'll be a string it's probably OK. Your choice :) 
```suggestion
	const rootId = getBlockRootClientId( clientId );
	if ( ! isString( rootId ) ) {
		return defaultView;
	}
```